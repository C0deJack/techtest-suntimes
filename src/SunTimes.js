import { useEffect, useState } from 'react';
import styled from 'styled-components';
import getLocation from './getLocation';
import getSunTimes from './getSunTimes';

const StyledSunTimes = styled.div`
    padding: 0.5rem;
`;

export default function SunTimes() {
    const [location, setLocation] = useState({});

    useEffect(() => {
        console.log('useEffect - getting location');
        setLocation(getLocation());
    }, []);

    const [sunTimes, setSunTimes] = useState('');

    useEffect(() => {
        getSunTimes(location)
            .then(data => {
                console.log(data);
                setSunTimes(data);
            })
            .catch(err => console.log(err));
    }, [location]);

    return (
        <>
            {sunTimes ? (
                <StyledSunTimes title='sunrise and sunset'>
                    <div>{sunTimes.sunrise}</div>
                    <div>{sunTimes.sunset}</div>
                </StyledSunTimes>
            ) : (
                <div>Loading</div>
            )}
        </>
    );
}
