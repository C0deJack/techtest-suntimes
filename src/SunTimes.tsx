import { ReactElement, useEffect, useState } from 'react';
import styled from 'styled-components';
import getLocation from './getLocation';
import getSunTimes, { SunTimesType } from './getSunTimes';
import sunriseIcon from './assets/sunrise.png';
import sunsetIcon from './assets/sunset.png';
import Loading from './loading';
import Time from './Time';
import BackgroundImage from './BackgroundImage';

const StyledSunTimes = styled.div`
    overflow: hidden;
    position: relative;
    height: 100vh;
    width: 100vw;
    min-width: 100%;
    color: #000;

    div.sunrise {
        background-color: rgba(255, 255, 255, 0.7);
        margin: 0;
        position: absolute;
        top: 50%;
        left: 50%;
        -ms-transform: translate(-50%, -50%);
        transform: translate(-50%, -50%);
        width: 60vw;
        min-width: 230px;
        border-radius: 2rem;
        padding: 3rem;
    }

    hr {
        height: 2px;
        background-color: #000;
    }

    @media (min-width: 576px) {
        div.sunrise {
            width: 50vw;
        }
    }
`;

export default function SunTimes(): ReactElement {
    const [location, setLocation] = useState({
        lat: '',
        lon: '',
    });

    useEffect(() => {
        getLocation(setLocation);
    }, []);

    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    const [sunTimes, setSunTimes] = useState<SunTimesType | '' | any>('');

    useEffect(() => {
        getSunTimes(location)
            .then(data => {
                setSunTimes(data);
            })
            .catch(err => console.log(err));
    }, [location]);

    return (
        <StyledSunTimes>
            <div className='sunrise' title='sunrise and sunset'>
                {sunTimes ? (
                    <>
                        <Time
                            title={'Sunrise'}
                            icon={sunriseIcon}
                            time={sunTimes.sunrise}
                        />
                        <hr />
                        <Time
                            title={'Sunset'}
                            icon={sunsetIcon}
                            time={sunTimes.sunset}
                        />
                    </>
                ) : (
                    <Loading />
                )}
            </div>

            {sunTimes && <BackgroundImage sunTimes={sunTimes} />}
        </StyledSunTimes>
    );
}
