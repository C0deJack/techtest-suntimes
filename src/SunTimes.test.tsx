import renderer from 'react-test-renderer';
import SunTimes from './SunTimes';

test('SunTimes renders correctly', () => {
    const sunTimes = renderer.create(<SunTimes />).toJSON();
    expect(sunTimes).toMatchSnapshot();
});
