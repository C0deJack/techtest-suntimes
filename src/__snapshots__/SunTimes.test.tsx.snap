// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SunTimes renders correctly 1`] = `
<div
  className="sc-dkPtRN iOczpI"
>
  <div
    className="sunrise"
    title="sunrise and sunset"
  >
    <div>
      <h3>
        LOADING
      </h3>
      <img
        alt="loading icon"
        src="loading.svg"
      />
    </div>
  </div>
  
</div>
`;
